insights/core/spec_factory.py

```diff
@@ -932,7 +932,11 @@ class first_of(object):
     """
     def __init__(self, deps):
         self.deps = deps
-        self.raw = deps[0].raw
+        self.raw = False
+        for c in deps:
```

Comment:
@vdymna - I'm sorry about missing the necessary comments for this change. 

Like the original code, it will get the `"raw"` attribute of the first item of the `first_of`, however, since the `first_of` for the new `certificates_info` spec is defined as:
```
    certificates_info = first_of(
        [
            certificates.cert_and_path,
            # To support the old insights-client without core collection
            simple_command("/usr/bin/find /etc/origin/node /etc/origin/master /etc/pki /etc/ipa /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem /etc/rhsm/ca/katello-default-ca.pem /etc/rhsm/ca/katello-default-ca.pem -type f -exec /usr/bin/openssl x509 -noout -dates -issuer -subject -in '{}' \; -exec echo 'FileName= {}' \;", keep_rc=True)
        ]
    )
```
Its first item is the `certificates.cert_and_path` which is a datasource function but does not have such a `raw` attribute.
To avoid the following `AttributeError`, I made this change: to get the `raw` attribute from the first item which does have such an attribute, otherwise set its value as `False`. How do you think?
```
insights/specs/default.py:132: in DefaultSpecs
    certificates_info = first_of(
insights/core/spec_factory.py:935: in __init__
    self.raw = deps[0].raw
E   AttributeError: 'function' object has no attribute 'raw'
```
I will add such a comment to the code at first, Please correct me, if there is anything wrong. Thanks.

Reply:
That makes sense now, thank you for the explanation.